source-code/core/app/src/createInlang.ts

```diff
@@ -0,0 +1,220 @@
+import { ImportFunction, ResolveModulesFunction, resolveModules } from "@inlang/module"
+import { NodeishFilesystemSubset, InlangConfig, Message, tryCatch } from "@inlang/plugin"
+import { TypeCompiler } from "@sinclair/typebox/compiler"
+import { Value } from "@sinclair/typebox/value"
+import { ConfigPathNotFoundError, ConfigSyntaxError, InvalidConfigError } from "./errors.js"
+import { LintRuleThrowedError, LintReport, lintMessages } from "@inlang/lint"
+import { createRoot, createSignal, createEffect } from "./solid.js"
+import { createReactiveQuery } from "./createReactiveQuery.js"
+
+const ConfigCompiler = TypeCompiler.Compile(InlangConfig)
+
+/**
+ * Creates an inlang instance.
+ *
+ * - Use `_import` to pass a custom import function for testing,
+ *   and supporting legacy resolvedModules such as CJS.
+ *
+ */
+export const createInlang = async (args: {
+	configPath: string
+	nodeishFs: NodeishFilesystemSubset
+	_import?: ImportFunction
+}): Promise<InlangInstance> => {
+	return await createRoot(async () => {
+		const [initialized, markInitAsComplete, markInitAsFailed] = createAwaitable()
+
+		// -- config ------------------------------------------------------------
+
+		const [config, setConfig] = createSignal<InlangConfig>()
+		createEffect(() => {
+			//console.log("set config internal")
+			loadConfig({ configPath: args.configPath, nodeishFs: args.nodeishFs })
+				.then((config) => {
+					setConfig(config)
+				})
+				.catch((err) => {
+					//console.error("Error in load config ", err)
+					markInitAsFailed(err)
+				})
+		})
+		// TODO: create FS watcher and update config on change
+
+		// -- resolvedModules -----------------------------------------------------------
+
+		const [resolvedModules, setResolvedModules] =
+			createSignal<Awaited<ReturnType<ResolveModulesFunction>>>()
+
+		createEffect(() => {
+			const conf = config()
+			if (!conf) return
+			loadModules({ config: conf, nodeishFs: args.nodeishFs, _import: args._import })
+				.then((resolvedModules) => {
+					setResolvedModules(resolvedModules)
+					// TODO: handle `detectedLanguageTags`
+				})
+				.catch((err) => {
+					delete err.stack // TODO: find a way to get rid of the filename in the stack trace or else we won't be able to read the actual error message
+					console.error("Error in load modules ", err)
+				})
+		})
+		// -- messages ----------------------------------------------------------
+
+		let configValue: InlangConfig
+		createEffect(() => configValue = config()!) // workaround to not run effects twice (e.g. config change + modules change) (I'm sure there exists a solid way of doing this, but I haven't found it yet)
+
+		const [messages, setMessages] = createSignal<Message[]>()
+		createEffect(() => {
+			const _resolvedModules = resolvedModules()
+			if (!_resolvedModules) return
+
+			makeTrulyAsync(
+				_resolvedModules.data.plugins.data.loadMessages({ languageTags: configValue!.languageTags }),
```

Comment:
should we output a warning if `loadMessages` does not return anything?

Reply:
prob push an error to `errors` no?

Reply:
It isn't really an error or is it?

Reply:
Are you referring to empty array or undefined with "if does not return anything"? 

`loadMessages` must return an array of messages. If `undefined` is returned, I would consider the falsy return value a plugin error

Reply:
`undefined` => error
empty array => warning that the plugin is probably not set up correctly

Reply:
I see. I would not console.warn something (will be ignored). Apps should handle "hey, you don't have messages yet." as part of the app flow.